tests/main/network-retry/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: Ensure network retry works correctly
+
+prepare: |
+    echo "Break DNS"
+    mv /etc/resolv.conf /etc/resolv.conf.save
```

Comment:
is having the local resolver not running at all equivalent to this or is that a different case?

Reply:
I get exactly the same error message when I move away resolv.conf - I think this is what is actually happening on these devices: /etc/resolv.conf is not setup yet and snapd (via golang runtime) tries to use it. But to know for sure we need better debug logs from the devices that fail.

Reply:
My point its, if you want to have a reliable test, please include nsswitch in the mix.